src/site/content/en/lighthouse-performance/total-blocking-time/index.md

```diff
@@ -0,0 +1,92 @@
+---
+layout: post
+title: Total Blocking Time
+description: |
+  Learn about Lighthouse's Total Blocking Time metric and
+  how to measure and optimize it.
+web_lighthouse:
+  - total-blocking-time
+date: 2019-10-09
+---
+
+Total Blocking Time (TBT) is one of the metrics tracked in the **Performance** section
+of the Lighthouse report. Each metric captures some aspect of page load speed.
+
+The Lighthouse report displays TBT in milliseconds:
+
+<figure class="w-figure">
+  <img class="w-screenshot" src="total-blocking-time.jpg" alt="A screenshot of the Lighthouse Total Blocking Time audit">
+</figure>
+
+## What TBT measures
+
+TBT measures the total amount of time that a page is blocked from responding to user input,
+such as mouse clicks, screen taps, or keyboard presses. The sum is calculated by adding
+the duration of all [long tasks] between [First Contentful Paint][fcp] and [Time to Interactive][tti].
+Any task that executes for more than 50 milliseconds is a long task.
+
+## How Lighthouse determines your TBT score
+
+Your TBT score is a comparison of your page's TBT time and TBT times for the top 10,000 sites
+when loaded on mobile devices. The top site data includes 404 pages.
+
+This table shows how to interpret your TBT score:
+
+<div class="w-table-wrapper">
+  <table>
+    <thead>
+      <tr>
+        <th>TBT time<br>(in milliseconds)</th>
+        <th>Color-coding</th>
+      </tr>
+    </thead>
+    <tbody>
+      <tr>
+        <td>0–200</td>
+        <td>Green (fast)</td>
+      </tr>
+      <tr>
+        <td>200-600</td>
+        <td>Orange (average)</td>
+      </tr>
+      <tr>
+        <td>Over 600</td>
+        <td>Red (slow)</td>
+      </tr>
+    </tbody>
+  </table>
+</div>
+
+{% include 'content/lighthouse-performance/scoring.njk' %}
+
+## How to improve your TBT score
+
+See [What is causing my long tasks?](/long-tasks-devtools/#what-is-causing-my-long-tasks) to learn
+how to diagnose the root cause of long tasks with the Performance panel of Chrome DevTools.
+
+In general, the most common causes of long tasks are:
+
+* Inefficient JavaScript statements. For example, after analyzing your code in the Performance panel, suppose
+  you see a call to `document.querySelectorAll('a')` that returns 2000 nodes. Refactoring your code to
```

Comment:
this specific example of the nodes is something I'm not super confident is common, maybe we could replace with a reference to force reflows or something instead? i.e. "instead of setting style and measuring in a loop over all 2000 nodes, measure all nodes once and then set the styles second" or something

https://developers.google.com/web/fundamentals/performance/rendering/avoid-large-complex-layouts-and-layout-thrashing#avoid_layout_thrashing perhaps?

Reply:
I got the querySelectorAll example from Addy's post: https://web.dev/long-tasks-devtools/#what-is-causing-my-long-tasks

I can change this if you feel strongly about it but I think the simple example gets the idea across. Per your other comment this also doesn't seem like a big topic to focus on (but we should probably still mention just in case).

Reply:
Ah OK, no worries this works then the simplicity *is* nice! My other comment was the more important piece of this deal :)